src/Homepage.js

```diff
@@ -1,18 +1,43 @@
-import { useState, useEffect } from "react";
+import { useState, useEffect, useRef } from "react";
+
 
 const Homepage = () => {
     const [amountOfSnippets, setamountOfSnippets] = useState('')
     const [text, setText] = useState('')
     const [failed, setFailed] = useState('')
-    const url = "/api/generateText";
+    const [soundText, setSoundText] = useState('')
+    const [button, setButton] = useState('')
+    const [muted, setMuted] = useState(false)
+
+    const url = useRef("/api/generateText");
```

Comment:
No need to useRef for url, you can just hard code it in the fetch it's self, this line is useless